memory/org.eclipse.cdt.debug.ui.memory.search/src/org/eclipse/cdt/debug/ui/memory/search/MemorySearchResultsPage.java

```diff
@@ -173,34 +174,48 @@ public Object[] getElements(Object inputElement) {
 
 			@Override
 			public void selectionChanged(final SelectionChangedEvent event) {
-				if (event.getSelection() instanceof StructuredSelection) {
-					IMemoryRenderingContainer containers[] = ((IMemorySearchQuery) fQuery).getMemoryView()
-							.getMemoryRenderingContainers();
+				IMemoryRenderingContainer containers[] = ((IMemorySearchQuery) fQuery).getMemoryView()
+						.getMemoryRenderingContainers();
+				if (containers == null || containers.length == 0)
+					return;
+
+				if (event.getSelection() instanceof StructuredSelection
```

Comment:
If we're rewriting this if anyway, lets use modern Java to simplify it. (I can't use GitHub suggestions as the change crosses unmodified parts of the code:

These two lines:

```java
				if (event.getSelection() instanceof StructuredSelection sel
						&& sel.getFirstElement() instanceof MemoryMatch match) {
```

replace everything including the `if (match == null) return;`

Reply:
You picked up a small part of this change, but not the whole thing, for example the if match == null is never true and there is a warning now on that.

I am going to split the code cleanup commit from the new behaviour (double-click) commit and update this PR.